test/realm_test.dart

```diff
@@ -1756,6 +1755,125 @@ Future<void> main([List<String>? args]) async {
       });
     }
   }
+  test('Realm.refresh no changes', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    final result = realm.refresh();
+    expect(result, false);
+  });
+
+  test('Realm.refresh into sync transaction', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    var called = false;
+    bool isRefreshed = false;
+    final transaction = realm.beginWrite();
+    realm.refreshAsync().then((refreshed) {
+      called = true;
+      isRefreshed = refreshed;
+    });
+    realm.add(Person("name"));
+    transaction.commit();
+
+    await Future<void>.delayed(Duration(milliseconds: 1));
+    expect(isRefreshed, false, reason: "There are no new data");
```

Comment:
```suggestion
    expect(isRefreshed, false, reason: "There is no new data");
```

Reply:
I think the reason should state why we have this expecations. So this should be 
"There should be no new data"

Reply:
In general adding a reason like this here is not adding much info. I see using a `reason` when the expectation is not clear enough